File: 8_StringtoInteger/8_v1.py
class Solution(object):
    def myAtoi(self, str):
        """
        :type str: str
        :rtype: int
        """
        INT_MAX = 2147483647
        INT_MIN = -2147483648
        
        m = re.match("^ *([\\+-]?[0-9]+).*$", str)
        
        if not m:
            return 0
        
        result = int(m.group(1))
        
        return min(result, INT_MAX) if result > 0 else max(result, INT_MIN)

File: 8_StringtoInteger/8_v1.js
/**
 * @param {string} str
 * @return {number}
 */
var myAtoi = function(str) {
    var INT_MAX = 2147483647
    var INT_MIN = -2147483648
    
    var m = /^\s*([\+-]?\d+).*$/.exec(str)
    
    if (m === null)
        return 0
        
    var result = parseInt(m[0])
    
    return result > 0? Math.min(result, INT_MAX) : Math.max(result, INT_MIN)
};
